api/src/controllers/login.js

```diff
@@ -285,23 +291,23 @@ module.exports = {
       })
       .catch(next);
   },
-  post: (req, res) => {
+  post: async (req, res) => {
+    const limited = await isRateLimited(req);
+    if (limited) {
+      return serverUtils.rateLimited(req, res);
+    }
```

Comment:
Is this not automatically handled by the middleware?

Reply:
The problem is the body parser doesn't run before the middleware so we don't have access to the user/pass in the JSON body until now...

Reply:
oh, bummer! thanks for explaining